pkg/middleware/auth.go

```diff
@@ -47,7 +47,11 @@ func notAuthorized(c *m.ReqContext) {
 		return
 	}
 
-	WriteCookie(c.Resp, "redirect_to", url.QueryEscape(c.Req.RequestURI), 0, newCookieOptions)
+	redirectTo := c.Req.RequestURI
+	if setting.AppSubUrl != "" && !strings.HasPrefix(redirectTo, setting.AppSubUrl) {
```

Comment:
> Why doesn't c.Req.RequestURI include AppSubUrl?

As far as I understand, the user can request whatever URL.
For instance, the user can request `http://localhost:3000` even though the subpath is `/grafana`.

In addition to this, we always prepended the subpath before [this](https://github.com/grafana/grafana/commit/a7f96ea98e34da8253f7bb071ae6e2bf7435e7b0#diff-c6731ee8c1631796c076df9ae9e4d2c7) change which was introduced for fixing [this](https://github.com/grafana/grafana/pull/21652#issuecomment-582601735) problematic situation.

Probably now that we have validation for the redirects (and if it's working correctly) we no more need to prepend it
because [invalid redirect paths are ignored](4e52076b9dc8c312b2b9323621274a6b0de45601).
But I think it's safer to keep it here, since it appends it only if it's missing.